### PR TITLE
Add doxygen into getgb2.f, Part of #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Part of #55 
Doxygen has been added into getgb2.f. It was shown on:
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/getgb2_8f.html

### CODE REVIEW

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
Doxygen does not recognize the "%" in your comments. Try adding a backslash before the %

**Comment by Hang-Lei-NOAA** [comment]
@GeorgeGayno-NOAA 
As you suggested, it works! Thanks!

**Comment by Hang-Lei-NOAA** [comment]
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/getgb2_8f.html

File: src/getgb2.f [hunk 1]
```diff
@@ -1,291 +1,235 @@
...
+C>    - gfld\%fld Array of gfld\%ndpts unpacked data points. This element
+C>    is actually a pointer to an array that holds the data.
+C>    @param[out] IRET integer return code
+C>    - 0 all ok
+C>    - 96 error reading index
+C>    - 97 error reading grib file
+C>    - 99 request not found
+C>    - other gf_getfld grib2 unpacker return code
+C>
+C>    Subprograms called:
```
Comment (edwardhartnett) [inline src/getgb2.f]: You can remove the section "subprograms called" everywhere you see it. Doxygen has a neat feature we can turn on which automatically provides a list of all subprograms called, which we then don't have to maintain. (If we are maintaining it, it inevitably will become wrong over time, whereas doxygen figures it out automatically each time and is always correct).

**Review by edwardhartnett — APPROVED** [review]

File: src/getgb2.f (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply src/getgb2.f]: Thanks for the knowledge!

**Review by edwardhartnett — APPROVED** [review]

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**